Sources/SwiftDocC/Utility/FoundationExtensions/String+Capitalization.swift

```diff
@@ -0,0 +1,34 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+extension StringProtocol {
+    
+    /// Returns the string with the first letter capitalized.
+    /// This auto-capitalization only occurs if the first word is all lowercase and contains only characters A-Z.
+    /// The first word can also contain punctuation (e.g. a period, comma, hyphen, semi-colon, colon).
+    func capitalizeFirstWord() -> String {
+        
+        let firstWord = self.prefix(while: { !$0.isWhitespace && !$0.isNewline })
+        
+//        let firstWord = self.components(separatedBy: " ").first ?? ""
```

Comment:
It seems like it's not possible to rewrite this function so that it returns a `StringProtocol`, because it isn't possible to work with substrings in StringProtocol vs string. However, I refactored this so that the extension is on String instead of StringProtocol, so that we are working with `self` instead of `"\(self)"` as I had it before.